Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,266 @@
+`(k, a) ↦ (k, k • 1 + ContinuousLinearMap.mul 𝕜 A a)`. It turns out that among all norms on the
+unitization satisfying the properties specified above, the norm inherited from `PiLp 1` is maximal,
+and the norm inherited from this pullback is minimal.
+
+For possibly non-unital `RegularNormedAlgebra`s  `A` (over `𝕜`), we construct a `NormedAlgebra`
+structure on `Unitization 𝕜 A` using the pullback described above. The reason for choosing this norm
+is that for a C⋆-algebra `A` its norm is always regular, and the pullback norm on `Unitization 𝕜 A`
+is then also a C⋆-norm.
+
+## Main definitions
+
+- `NonUnitalAlgHom.Lmul : A →ₙₐ[𝕜] A →L[𝕜] A`: `ContinuousLinearMap.mul` upgraded to a non-unital
+  algebra homomorphism.
+- `Unitization.leftRegRep : Unitization 𝕜 A →ₐ[𝕜] (𝕜 × (A →L[𝕜] A))`: The first coordinate of this
+  map is just `Unitization.fst` and the second is the left regular representation of
+  `Unitization 𝕜 A` *acting on `A`*. This is the `Unitization.lift` of `NonUnitalAlgHom.Lmul`.
+  We use this map to pull back the `NormedRing` and `NormedAlgebra` structures.
+
+## Implementation details
+
+We ensure that the uniform structure, and hence also the topological structure, is definitionally
+equal to `instUniformSpaceProd` (viewing `Unitization 𝕜 A` as `𝕜 × A`) by means of forgetful
+inheritance. The same is true of the bornology.
+
+-/
+
+variable (𝕜 A : Type _) [NontriviallyNormedField 𝕜] [NonUnitalNormedRing A]
+variable [NormedSpace 𝕜 A] [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A]
+
+open ContinuousLinearMap
+
+/-- Multiplication on the left in a non-unital algebra `A` as a non-unital algebra homomorphism
+into the algebra of *continuous* linear maps. This has more algebraic structure than
+`ContinuousLinearMap.mul`, but there is no longer continuity bundled in the first coordinate. -/
+noncomputable def NonUnitalAlgHom.Lmul : A →ₙₐ[𝕜] A →L[𝕜] A :=
+  { mul 𝕜 A with
+    toFun := fun a => mul 𝕜 A a
+    map_mul' := fun a b => by ext x; simp [mul_assoc a b x]
+    map_zero' := by ext x; simp only [map_zero] }
+
+variable {𝕜 A}
+
+@[simp]
+theorem NonUnitalAlgHom.coe_Lmul : ⇑(NonUnitalAlgHom.Lmul 𝕜 A) = mul 𝕜 A :=
+  rfl
+
+variable (𝕜 A)
+
+namespace Unitization
+
+/-- `leftRegRep` stands for "left regular representation" which is multiplication on the left. So,
+given `(k, a) : Unitization 𝕜 A`, the second coordinate of `Unitization.leftRegRep (k, a)` should
+be the representation of `Unitization 𝕜 A` on `A →L[𝕜] A`; note that this is not just
+`NonUnitalAlgHom.Lmul` for a few reasons: (a) that would either be `A` acting on `A`, or
+(b) `Unitization 𝕜 A` acting on `Unitization 𝕜 A`, and (c) that's a `NonUnitalAlgHom` but
+here we need an `AlgHom`. In addition, the first coordinate of `Unitization.leftRegRep (k, a)`
+should just be `k`. See `Unitization.leftRegRep_apply` also. -/
+noncomputable def leftRegRep : Unitization 𝕜 A →ₐ[𝕜] 𝕜 × (A →L[𝕜] A) :=
+  (lift 0).prod (lift <| NonUnitalAlgHom.Lmul 𝕜 A)
```

Comment:
Is this terminology standard? I had in mind that the left regular representation of an algebra was just the multiplication on itself, but maybe that is only satisfying for unital algebras.

Reply:
No, it's not standard. I'm open to other names, especially because the current name doesn't take the first coordinate into account either. Maybe `liftLeftRefRep`?

Reply:
What do you think of `splitMul`? Because it's exactly the multiplication by an element of `Unitization 𝕜 A`, but splitted as an action on `𝕜` and an action on `A`.

Another option, as suggested by [this answer on MO](https://mathoverflow.net/a/210217), is to actually consider the mutliplication as a map `Unitization 𝕜 A →ₐ[𝕜] (𝕜 × A) →L[𝕜] (𝕜 × A)`, which should give the same norm from what I understand. But that may be a bit more work, and I'm not sure how useful that is.

Reply:
@ADedecker `splitMul` is fine. However, as for the answer on MO, I believe my construction is nicer. In the MO version, you must first bundle the function `fun a ↦ (0 : 𝕜 →L[𝕜] 𝕜).prodMap (ContinuousLinearMap.mul 𝕜 A a)` from `A` to `(𝕜 × A) →L[𝕜] (𝕜 × A)` into a non-unital algebra homomorphism, called `foo 𝕜 A`, then the algebra homomorphism described on MO is `Unitization.lift (foo 𝕜 A)`. I really don't think this any better, and in fact, I think it would be significantly harder to work with. And just to be clear, it's not like the MO construction avoids the need for `ContinuousLinearMap.mul 𝕜 A` to be injective, that is still necessary to ensure injectivity of this algebra homomorphism (as should be clear from the above).

Reply:
I understand my mistake: I thought that `foo 𝕜 A` was exactly mutiplication in the unitization (which is clearly injective), but there's a term missing. So indeed this doesn't simplify anything.